src/workloads/query/PercentilesSpillToDisk.yml

```diff
@@ -0,0 +1,158 @@
+
+Keywords:
+- setWindowFields
+- percentile
+- spill
+
+AutoRun:
+- When:
+    mongodb_setup:
+      $eq:
+      - standalone-all-feature-flags
+    branch_name:
+      $neq:
+      - v4.0
+      - v4.2
+      - v4.4
+      - v5.0
+      - v6.0
+
+GlobalDefaults:
+  MaxPhases: &maxPhases 8
+  Database: &db percentiles
+  Collection: &coll Collection0
+
+ActorTemplates:
+- TemplateName: ComputePercentile
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: "ComputePercentile"}}
+    Type: CrudActor
+    Threads: 1
+    Phases:
+      OnlyActiveInPhases:
+        Active: [{^Parameter: {Name: "OnlyActiveInPhase", Default: 1024}}]
+        NopInPhasesUpTo: *maxPhases
+        PhaseConfig:
+          Repeat: 10
+          Database: *db
+          Collection: *coll
+          Operations:
+          - OperationName: aggregate
+            OperationCommand:
+              Pipeline: [
+                {
+                  $setWindowFields:
+                    {
+                      partitionBy: {^Parameter: {Name: "Partition", Default: null}},
+                      sortBy: {^Parameter: {Name: "Sort", Default: {not_spill_indexed: 1}}},
+                      output: {
+                        p: {
+                          $percentile: {p: [0.75], input: "$data", method: "approximate" },
+                          window: { documents: [0, "unbounded"] }
+                        }
+                      }
+                    }
+                }
+              ]
+              Options:
+                AllowDiskUse: {^Parameter: {Name: "Disk", Default: true}}
+
+Actors:
+- Name: CreateDataset
+  Type: Loader
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *maxPhases
+      PhaseConfig:
+        Threads: 1
+        Repeat: 1
+        Database: *db
+        CollectionCount: 1
+        # We have 100k documents with each being between 10kB to 20kB. When divided into 20 partitions, it won't spill, but when divided into fewer partitions (like 2) it will spill.
+        DocumentCount: 100000
+        BatchSize: 1000
+        Document:
+          # We will use the _id field to create evenly distributed partition keys to test spilling.
+          _id: {^Inc: {start: 0}}
+          # This data field is what we calculate the percentiles of.
+          data: {^RandomDouble: {distribution: normal, mean: 50.0, sigma: 10.0}}
+          # This string pads the doc size out to around 10KB.
+          string: &string {^RandomString: {length: 10000}}
+
+- Name: CreatePartitionAndIndex
+  Type: RunCommand
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *maxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *db
+        Operations:
+        - OperationName: RunCommand
+          OperationCommand:
+            update: *coll
+            updates: [
+              {
+                q: {},
+                u: [
+                  {$set: {
```

Comment:
I suspect it might be faster to create the dataset with these additional fields at the start, rather than adding them through and update operation.

Reply:
Done! I was concerned about having the partitions even, but that isn't necessary because they'll still be large enough to spill even if they aren't perfectly even.